thrift/server.go

```diff
@@ -94,13 +94,33 @@ func (s *Server) SetContextFn(f func(ctx context.Context, method string, headers
 	s.ctxFn = f
 }
 
-func (s *Server) onError(err error) {
+func (s *Server) onError(ctx context.Context, method string, err error) {
 	// TODO(prashant): Expose incoming call errors through options for NewServer.
-	// Timeouts should not be reported as errors.
+	logger := s.log.WithFields(
+		tchannel.ErrField(err),
+		tchannel.LogField{Key: "method", Value: method},
+	)
+
+	// Add additional context about the remote peer if available.
+	if call := tchannel.CurrentCall(ctx); call != nil {
+		remotePeer := call.RemotePeer()
+		logger = logger.WithFields(
+			tchannel.LogField{Key: "callerName", Value: call.CallerName()},
+
+			// TODO: These are very similar to the connection fields, but we don't
+			// have access to the connection's logger. Consider exposing the
+			// connection through CurrentCall.
```

Comment:
You have the InboundCall available where you call this method, and InboundCall has access to the connection. Can you not use that?

Reply:
The inbound call has access to the connection in an unexported field, but since we're not in the same package, we can't access it.

However, the `call` has everything that I was getting out of the context, so simplified that logic a little.